### ISSUE TITLE
Functions do not properly type check arguments.

### ISSUE THREAD

**Issue by DavidKnott** [ethereum/viper#331]
Currently functions do not properly check the types of their input arguments and do not error when not enough arguments are given:

    age:public(num)

    def __init__(start_age: num):
        self.age = start_age

    def set_age(new_age: num):
        self.age = new_age

The above function doesn't fail when it's initialized with the following
```
get_contract(source_code, args=[])   # age is set to 0
get_contract(source_code, args=['b'])  # age is set to 98
```
After it's initialized the following can be called without errors:
```
c.set_age()  # age is set to 0
c.set_age('b')  # age is set to 98
```

**Comment by fubuloubu** [comment]
Why'd you close this issue @DavidKnott?

**Comment by DavidKnott** [comment]
I checked out Solidity and it doesn't have function argument type checking functionality either, I believe this is due to the EVM storing all data in byte code.  As far as I know there's no easy way to do type checking on the EVM side of things (e.g. b and 98 are the same in bytes).  So I closed this issue because although I think how it works is dangerous I think it's an issue more with the EVM then Viper.
